wordpress-mobile/AztecEditor-iOS
# Refactor TextAttachment to be MediaAttachment.

Refs #165 

This PR refactor TextAttachment to be MediaAttachment. It then creates a ImageAttachment to replace the uses of the previous TextAttachment and it also implements VideoAttachment as a subclass of MediaAttachment.

This change allow sharing of a lot of layout and delegate code between the Image and Video implementations and make simples in the future do changes to both systems.

How to test:
 * Run the Demo app
 * Check that the image and video of the demo show up correctly
 * Test the interactions with the image ( add image, remove image, tap for details)
 * Test the interactions with videos ( tap to play, remove)
